src/mon/AuthMonitor.cc

```diff
@@ -1850,14 +1829,7 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
 	}
       }
 
-      KeyRing kr;
-      kr.add(entity, entity_auth.key);
-      if (f) {
-	kr.set_caps(entity, entity_auth.caps);
-	kr.encode_formatted("auth", f.get(), rdata);
-      } else {
-	kr.encode_plaintext(rdata);
-      }
+      _encode_key(entity, entity_auth, rdata, f.get());
```

Comment:
OOPS!

`caps` is not specified, so `nullptr` will cause to miss the `kr.set_caps` call.

Reply:
Fixed.